src/_pytest/assertion/util.py

```diff
@@ -96,6 +100,23 @@ def _format_lines(lines: Sequence[str]) -> List[str]:
     return result
 
 
+def _get_number_of_calls(func_name: str) -> int:
+    """
+    Get the number of calls for the specified function.
+    To get this, the stack is going to be inspected.
+
+    :param str func_name: the name of the function
+    :return: number of calls from stack
+    :rtype: int
+    :raises ValueError: if the func_name was not found in the stack.s
+    """
+    functions = [frame.function for frame in inspect.stack()]
+    func_calls = Counter(functions).get(func_name, 0)  # int
```

Comment:
Generally it is possible for multiple different functions in the stack to have the same name, in which case this will get confused.